### ISSUE TITLE
Bad user experience when server can't fixpoint

### ISSUE THREAD

**Issue by cmontella** [witheve/Eve#411]
Here is a program that repros:

Initialize the board
```
match
  [#session-connect]
commit
  [@board n: 3, cell-size: 50, player: "X"]
```

Give the board some cells
```
match
  board = [@board n]
  i = range[from: 0, to: n]
  j = range[from: 0, to: n]
bind
  board.cells += [#cell row: i, col: j, player: ""]
```

Do some things when we click
```
match
  [#click #direct-target element]
  [@board n player]
  cell = [#cell row: element.row col:element.col]
bind
  cell.player := player // Causes non-fixpoint situation
  [#div text: "({{cell.row}},{{cell.col}})"]
```

The offending line is commented in the block above. The reason the instability occurs is because we are binding player to "" in the second block, and binding it to another value in the last block.

The real issue here (pending non-fixpoint detection and prevention) is the user experience, which is that the server hangs, eventually crashes, and then the client loses all your work. A possible short term fix is to limit churning to ~100 iterations, and tell the user that the program can't make any progress.

**Comment by joshuafcole** [comment]
This guy falls under the umbrella of #412

**Comment by joshuafcole** [comment]
Closed along with #412. It's still not amazing, but its much better than it was, and improving it further will be an iterative process that proceeds in tandem with editor UX changes.
